### PR TITLE
[Chat] Fix the issue where file cannot be removed from edit box

### PR DESCRIPTION
**Pull request by jpeng-ms**
# What

Please note that this is an existing issue before the file sharing refactor:

Before (noticing how the UI is still in old design, this is last beta release without any file sharing refactor):

https://github.com/Azure/communication-ui-library/assets/109105353/ae9dfabc-f2e7-4bf2-994f-640a665f138d

After (latest change from main):

https://github.com/Azure/communication-ui-library/assets/109105353/90477058-abb9-4858-a864-6213ee1f9ef8

# Why

# How Tested
1. checkout this branch 
2. in chat sample app, provide the following settings:

```
 attachmentOptions: {
                uploadOptions: {
                  supportedMediaTypes: ['image/*', 'application/pdf'],
                  disableMultipleUploads: false,
                  handler: (files) => {
                    //files[0].notifyProgressChanged(0.5);
                    // noop
                    files[0].notifyCompleted({
                      name: 'ok.pdf',
                      extension: 'png',
                      progress: 1,
                      url: 'https://via.placeholder.com/150',
                      id: guidGenerator()
                    });
                  }
                }
}
```
3. start the app
4. attache a file and send the message
5. edit the message and remove attachment

# Process & policy checklist

- [X] I have updated the project documentation to reflect my changes if necessary.
- [X] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/chat-component-bindings/src/handlers/createHandlers.ts [hunk 1]
```diff
@@ -62,15 +62,21 @@ export const createDefaultChatHandlers = memoizeOne(
       onUpdateMessage: async (
         messageId: string,
         content: string,
+        /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */
```
Comment (vhuseinova-msft) [inline packages/chat-component-bindings/src/handlers/createHandlers.ts:65]: if options are under cc flag here, it shouldn't be available on line 71 for stable build

Comment (vhuseinova-msft) [reply packages/chat-component-bindings/src/handlers/createHandlers.ts:65]: not sure why stable build still pass...

File: packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMyMessageComponent.tsx [hunk 2]
```diff
@@ -128,7 +120,15 @@ export const ChatMyMessageComponent = (props: ChatMyMessageComponentProps): JSX.
         onSubmit={async (text, metadata, options) => {
           props.onUpdateMessage &&
             message.messageId &&
-            (await props.onUpdateMessage(message.messageId, text, metadata, options));
+            (await props.onUpdateMessage(
+              message.messageId,
+              text,
+              /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */
+              {
+                metadata: metadata ?? {},
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMyMessageComponent.tsx:128]: As `onUpdateMessage` func takes optionals, I'm wondering if we need to add empty object and array for these props

**Review by jimchou-dev — APPROVED** [review]

File: packages/chat-component-bindings/src/handlers/createHandlers.ts (hunk 1 above)
Comment (emlynmac) [reply packages/chat-component-bindings/src/handlers/createHandlers.ts:65]: Has `attachment-download` been added to the stabilized features list already and so this line goes through anyway?

Comment (jpeng-ms) [reply packages/chat-component-bindings/src/handlers/createHandlers.ts:65]: `not sure why stable build still pass...`
in stable, `options` is undefined, and therefore at line 71 `const updatedMetadata` would be set to `{}`


`Has attachment-download been added to the stabilized features list already and so this line goes through anyway?`
no it hasn't, jim will have a PR to decouple this later

**Review by vhuseinova-msft — APPROVED** [review]

**Pull request merged**